source/class.cpp

```diff
@@ -868,10 +868,22 @@ string bind_forward_declaration(CXXRecordDecl const *C, Context &context)
 
 	string const include = relevant_include(C);
 
+	string shared_ptr = "std::shared_ptr";
+	bool use_custom_shared = Config::get().use_custom_shared();
+	if( use_custom_shared )
+		shared_ptr = Config::get().custom_shared();
+
+
 	string c = "\t// Forward declaration for: " + qualified_name + " file:" + (include.size() ? include.substr(1, include.size() - 2) : "") + " line:" + line_number(C) + "\n";
 
-	string maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
-	if( is_inherited_from_enable_shared_from_this(C) ) maybe_holder_type = ", std::shared_ptr<{}>"_format(qualified_name);
+	string maybe_holder_type = ", {}<{}>"_format(shared_ptr, qualified_name);
+	//Check if the type is a custom shared pointer:
+	if( use_custom_shared && qualified_name.rfind(shared_ptr, 0) == 0 ) {
+		maybe_holder_type = "";
+	}
```

Comment:
Could you please elaborate what is going on here, why do we need this special case? Are we trying to restrict bindings of `custom_shared_ptr` class itself? If so then this check should rather go into https://github.com/RosettaCommons/binder/blob/master/source/type.cpp#L662

Reply:
Yes, what we want to avoid here is to create a `my_shared_ptr<my_shared_ptr<T>>`. If the `qualified_name` is already a custom shared pointer, we should not add a holder type associated to a custom shared pointer to a custom shared pointer.
Yes, I agree, it would be cleaner to have it with the other checks in https://github.com/RosettaCommons/binder/blob/master/source/type.cpp#L662 , sorry I was unaware of those.